### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by iK4tsu**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,59 +33,131 @@ struct Velocity
...
-	systemC(em.query!(Tuple!(Position,Velocity), Tuple!(With!Grounder, Without!Crawler)));
+	auto world = new EntityManagerT!(void delegate() @nogc);
+
+	world.onConstruct!int.connect((ref int i) {}); // @nogc callback
+	world.entity.emplace!int(4); // this won't be @nogc
+}
+```
+
+This happens because the internal code uses GC for allocations (still). However,
+`@safe pure nothrow` work like a charm.
```
Comment (FFY00) [inline README.md:148]:
```suggestion
`@safe pure nothrow` works like a charm.
```

File: README.md [hunk 2]
```diff
@@ -33,59 +33,131 @@ struct Velocity
...
+EntityManagerT can be used in `@safe pure nothrow` code and some parts are
+`@nogc`. What defines the majority of code safetyness is the Signal. Right now
+in D it is impossible to store functions with different attributes in one common
+type and still be able to track them. The common attribute to all functions is
+having none, which is the same as having `@system`. This forces all code that
+depends on it to be `@system` as well (which is almost the entire lib)
+
+The user can define their own safety restriction by passing a delegate with the
+attributes they desire for their callbacks. Those attributes influence the
+safetyness of almost the entire lib as explained above.
```
Comment (FFY00) [inline README.md:125]:
```suggestion
safetyness of almost the entire lib, as explained above.
```

File: README.md [hunk 3]
```diff
@@ -33,59 +33,131 @@ struct Velocity
...
+
+	// Select is a special Rule and can only be used as the first argument
+	// All other argument must be QueryRules 'With and Without'
+	static assert(!_traits(compiles, world.query!(Select!A, Select!B)));
 }
+```
+
+## Safety
+EntityManagerT can be used in `@safe pure nothrow` code and some parts are
+`@nogc`. What defines the majority of code safetyness is the Signal. Right now
```
Comment (FFY00) [inline README.md:117]:
```suggestion
`@nogc`. What defines the majority of code safeness is the Signal. Right now
```

File: README.md [hunk 4]
```diff
@@ -33,59 +33,131 @@ struct Velocity
...
+in D it is impossible to store functions with different attributes in one common
+type and still be able to track them. The common attribute to all functions is
+having none, which is the same as having `@system`. This forces all code that
+depends on it to be `@system` as well (which is almost the entire lib)
+
+The user can define their own safety restriction by passing a delegate with the
```
Comment (FFY00) [inline README.md:123]:
```suggestion
The user can define their own safety restrictions by passing a delegate with the
```

**Review by FFY00 — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (iK4tsu) [reply README.md:117]: Initially, I had that, but a quick Google search made me think this version existed. Now I searched it in the dictionary (as I should have the first time) and it does not.

File: README.md (hunk 2 above)
Comment (Ev1lbl0w) [inline README.md:125]: If I understood correctly, you refer to the code below, not above:
```suggestion
safetyness of almost the entire lib as explained below:
```

**Review by Ev1lbl0w — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (iK4tsu) [reply README.md:125]: No, it refers to the first paragraph. The code example demonstrates that sentence, and the last part of the same is referring to the explanation given in the previous paragraph.

File: README.md (hunk 4 above)
Comment (iK4tsu) [reply README.md]: done

File: README.md (hunk 3 above)
Comment (iK4tsu) [reply README.md]: done

File: README.md (hunk 2 above)
Comment (iK4tsu) [reply README.md]: done

File: README.md (hunk 1 above)
Comment (iK4tsu) [reply README.md]: done

**Pull request merged**